src/frontend/src/optimizer/plan_node/logical_join.rs

```diff
@@ -1338,6 +1338,10 @@ impl ToStream for LogicalJoin {
         {
             Ok(dynamic_filter)
         } else {
+            // Attempt to convert the inputs first to provide more accurate error reporting.
+            let _ = self.left().to_stream(ctx)?;
+            let _ = self.right().to_stream(ctx)?;
```

Comment:
Is this really necessary? I mean if the outer query is invalid in its form, seems the outer error is more important than inputs.

Reply:
Without these changes, the error message for some join with subqueries will become `nested loop join not supported".

> seems the outer error is more important than inputs

I'm not sure about this. 😕 The inner plan node resides the upstream of a job. If the upstream fails, then it does not matter whether the downstream can succeed, so does the error message.

Reply:
Do you have a test case that could trigger this error?  I think we can use a visitor to check whether `MaxOneRow` exists in `gen_optimized_logical_plan_for_stream` after subquery unnesting.

Reply:
You're right. Cannot remember well why this was necessary, maybe due to the reuse of `Limit`. It's not necessary now.